lib/checks/aria/required-children.js

```diff
@@ -62,13 +63,13 @@ function missingRequiredChildren(node, childRoles, all, role) {
 
 	// combobox exceptions
 	if (role === 'combobox') {
-		// remove 'textbox' from missing roles if combobox is a native text-type input
+		// remove 'textbox' from missing roles if combobox is a native text-type input or owns a 'searchbox'
 		var textboxIndex = missing.indexOf('textbox');
-		var textTypeInputs = ['text', 'search', 'email', 'url', 'tel'];
 		if (
 			textboxIndex >= 0 &&
-			node.nodeName.toUpperCase() === 'INPUT' &&
-			textTypeInputs.includes(node.type)
+			(getRole(node) === 'textbox' ||
+				owns(node, virtualNode, 'searchbox') ||
+				ariaOwns(ownedElements, 'searchbox'))
```

Comment:
I don't think you correctly identified the problem. This exception was correct the way it was. If the combobox is an input element with text input, an additional `textbox` is not required. The real problem is that we're not looking at inheritance in this rule.

In ARIA 1.1, `searchbox` inherits from `textbox`. So any `searchbox` that is owned, should count as a `textbox`. That's not too different from the issue #1677. A problem there is that `directory` inherits from `list`, and so any `listitem` element can be owned by a `directory`.

You'll probably have to ditch the `owns` and `ariaOwns` functions in this method to fix this problem. There's an `aria.getOwnedVirtual` method that I got started on a while ago, which I intended to at some point be a replacement of this code. We should maybe take the time to finish it. That way you don't have to run multiple QSA's for everything that inherits from another role.